backend/app/services/email/email_service_provider.py

```diff
@@ -1,12 +1,72 @@
+import os
+
+import boto3
+from botocore.exceptions import ClientError
+
+from app.interfaces.email_service import EmailContent, EmailTemplate
 from app.interfaces.email_service_provider import IEmailServiceProvider
 
 
 class AmazonSESEmailProvider(IEmailServiceProvider):
-    def __init__(self, aws_access_key: str, aws_secret_key: str):
-        pass
-
-    # TODO (Mayank, Nov 30th) - Create an email object to pass into this method
-    def send_email(
-        self, recipient: str, subject: str, body_html: str, body_text: str
-    ) -> dict:
-        pass
+    def __init__(
+        self,
+        aws_access_key: str,
+        aws_secret_key: str,
+        region: str,
+        source_email: str,
+        is_sandbox: bool = True,
+    ):
+        self.source_email = source_email
+        self.is_sandbox = is_sandbox
+        self.ses_client = boto3.client(
+            "ses",
+            region_name=region,
+            aws_access_key_id=aws_access_key,
+            aws_secret_access_key=aws_secret_key,
+        )
+
+        self.verified_emails = None
+        if self.is_sandbox:
+            response = self.ses_client.list_verified_email_addresses()
+            self.verified_emails = response.get("VerifiedEmailAddresses", [])
+
+    def _verify_email(self, email: str):
+        if not self.is_sandbox:
+            return
+        try:
+            if email not in self.verified_emails:
+                self.ses_client.verify_email_identity(EmailAddress=email)
+                print(f"Verification email sent to {email}.")
+        except Exception as e:
+            print(f"Failed to verify email: {e}")
+
+    def send_email(self, template: EmailTemplate, content: EmailContent) -> dict:
+        try:
+            self._verify_email(content.recipient)
+
+            self.ses_client.get_template(TemplateName=template.value)
+
+            template_data = content.data.get_formatted_string()
```

Comment:
nit: Imo you should group this with `self._verify_email(content.recipient)`, I got confused for a second reading the code given that `get_template` doesn't return anything right now.